src/components/NanoContract/NanoContractTransactionHeader.component.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import {
+  StyleSheet,
+  View,
+  TouchableWithoutFeedback,
+  Linking,
+  Text,
+} from 'react-native';
+import { t } from 'ttag';
+import { useSelector } from 'react-redux';
+import HathorHeader from '../HathorHeader';
+import { COLORS } from '../../styles/themes';
+import { combineURLs, getShortContent, getShortHash, getTimestampFormat } from '../../utils';
+import SimpleButton from '../SimpleButton';
+import { ArrowDownIcon } from '../Icons/ArrowDown.icon';
+import { TextValue } from '../TextValue';
+import { TextLabel } from '../TextLabel';
+import { TransactionStatusLabel } from '../TransactionStatusLabel';
+
+/**
+ * It presents the header of Nano Contract Transaction screen.
+ *
+ * @param {Obejct} props
+ * @param {Obejct} props.tx Transaction data
+ */
+export const NanoContractTransactionHeader = ({ tx }) => {
+  // TODO: use `toggleShrank` on 'onPress` of TouchableWithoutFeedback
+  const [isShrank, toggleShrank] = useState(false);
+
+  const isExpanded = () => !isShrank;
+
+  return (
+    <HathorHeader>
+      <HathorHeader.Central style={styles.headerCentral}>
+        <TouchableWithoutFeedback onPress={() => {}}>
+          <View style={styles.headerWrapper}>
+            <InfoContainer>
+              <TextValue title pb4>{getShortHash(tx.txId, 7)}</TextValue>
+              <TextLabel>{t`Transaction ID`}</TextLabel>
+            </InfoContainer>
+            {isShrank
+              && <HeaderShrank />}
+            {isExpanded()
+              && <HeaderExpanded tx={tx} />}
```

Comment:
Why don't you use a ternary here? It's the same but it seems a good use for it.

```
{ isShrank ? <HeaderShrank /> : <HeaderExpanded tx={tx} /> }
```

Reply:
Done.